src/usb_reset/src/ros_usb_reset.cpp

```diff
@@ -0,0 +1,89 @@
+#include "usb_reset/ros_usb_reset.hpp"
+
+int num = 0;
+
+ROS_Usb_Reset::ROS_Usb_Reset(QQuickItem * parent) :
+    QQuickPaintedItem(parent),
+    topic_value(),
+    ros_ready(false),
+    devices("no devices") {
+
+}
+
+void ROS_Usb_Reset::setup(ros::NodeHandle * nh, QQmlContext *ctxt) {
+
+    //TODO: make these parameters of the component
+      usb_sub = nh->subscribe(
+            topic_value.toStdString(),
+            1,
+            &ROS_Usb_Reset::receive_msg,
+            this
+      );
+      usb_pub = nh->advertise<std_msgs::Int16>(
+                    "rover/usb/reset",
+                    1,
+                    false
+            );
+
+    ros_ready = true;
+    curr_ctxt = ctxt;
+    ROS_INFO("Setup of usb component complete");
+}
+
+void ROS_Usb_Reset::receive_msg(const std_msgs::String::ConstPtr &msg) {
+    devices = msg->data;
```

Comment:
it seems weird that this would be a string - shouldn't it be an array of strings?